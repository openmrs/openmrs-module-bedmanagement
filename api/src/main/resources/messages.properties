${project.parent.artifactId}.title=bedmanagement Module
${project.parent.artifactId}.admissionLocations=Admission Locations
${project.parent.artifactId}.bedTypes=Bed Types
${project.parent.artifactId}.bedTags=Bed Tags