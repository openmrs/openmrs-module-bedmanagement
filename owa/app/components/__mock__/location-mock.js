Object.defineProperties(window.location, {
    host : {
        writable: true,
        value: '192.168.33.10'
    },
    hostname : {
        writable: true,
        value: '192.168.33.10'
    },
    href :{
        writable: true,
        value: 'https://192.168.33.10/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    origin :{
        writable: true,
        value: 'https://192.168.33.10'
    },
    pathname :{
        writable: true,
        value: '/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    port :{
        writable: true,
        value: ''
    },
    protocol :{
        writable: true,
        value: 'https:'
    }
});