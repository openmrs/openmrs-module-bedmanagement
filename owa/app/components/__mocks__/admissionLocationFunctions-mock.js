import pickBy from 'lodash.pickby';
import {admissionLocations, visitLocations, bedTypes} from 'utilities/__tests__/testData.json';

const admissionLocationFunctionsMock = {
    setActiveLocationUuid: (admissionLocationUuid) => jest.fn(),
    getActiveLocationUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    setState: (data) => jest.fn(),
    getState: () => jest.fn(),
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    getVisitLocations: () => {
        return visitLocations;
    },
    getBedTypes: () => {
        return bedTypes;
    },
    getAdmissionLocationByUuid: (admissionLocationUuid) => {
        return typeof admissionLocations[admissionLocationUuid] !== 'undefined' ?
            admissionLocations[admissionLocationUuid] : null;
    },
    getParentAdmissionLocation: (admissionLocationUuid) => {
        const parentAdmissionLocationUuid = admissionLocations[admissionLocationUuid].parentAdmissionLocationUuid;
        return parentAdmissionLocationUuid != null ? admissionLocations[parentAdmissionLocationUuid] : null;
    },
    getChildAdmissionLocations : (admissionLocationUuid) => {
        return  pickBy(admissionLocations, (admissionLocation) => {
            return admissionLocation.parentAdmissionLocationUuid == admissionLocationUuid;
        });
    },
    reFetchAllAdmissionLocations : () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default admissionLocationFunctionsMock;
