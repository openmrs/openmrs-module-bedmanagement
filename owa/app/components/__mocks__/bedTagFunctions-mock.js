import _ from 'lodash';

import {bedTags} from 'utilities/__tests__/testData.json';

const bedTagFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTags: () => {
        return bedTags;
    },
    getBedTagByUuid: (bedTagUuid) => {
        return _.find(bedTags, function (bedTag) {
            return bedTag.uuid == bedTagUuid;
        });
    },
    fetchBedTags: () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default bedTagFunctionsMock;
