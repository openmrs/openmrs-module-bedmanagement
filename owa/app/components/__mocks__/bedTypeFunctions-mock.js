import _ from 'lodash';

import {bedTypes} from 'utilities/__tests__/testData.json';

const bedTypeFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTypes: () => {
        return bedTypes;
    },
    getBedTypeById: (bedTypeId) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.id == bedTypeId;
        });
    },
    getBedTypeName: (bedTypeName) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.name == bedTypeName;
        });
    },
    fetchBedTypes: () => {
        return bedTypes;
    },
    notify : (notifyType, message) => jest.fn()
};

export default bedTypeFunctionsMock;
