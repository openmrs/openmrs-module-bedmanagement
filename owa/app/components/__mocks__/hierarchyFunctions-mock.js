import {admissionLocations} from 'utilities/__tests__/testData.json';

const hierarchyFunctionMock = {
    getActiveUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    setActiveUuid: (locationUuid) => jest.fn(),
    setState: (stateData) => jest.fn(),
    toggleIsOpen: (locationUuid, isOpen) => jest.fn(),
    setAdmissionLocationIsOpen: (locationUuid, isOpen) => jest.fn()
};

export default hierarchyFunctionMock;