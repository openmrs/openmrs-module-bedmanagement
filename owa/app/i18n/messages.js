export default {
    en: {
        ADMISSION_LOCATIONS: 'Admission Locations',
        BED_TYPES: 'Bed Types',
        BED_TAGS: 'Bed Tags',
        LOGOUT: 'Logout',
        CHOOSE_LANGUAGE: 'Choose language'
    }
};
