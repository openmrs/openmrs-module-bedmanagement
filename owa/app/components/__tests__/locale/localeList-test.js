import React from 'react';
import {mount, shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import {IntlProvider} from 'react-intl';

import LocaleList from 'components/locale/localeList';
import messages from 'i18n/messages';
require('components/__mocks__/location-mock');

const intlProvider = new IntlProvider({locale: 'en', messages: messages['en']}, {});
const {intl} = intlProvider.getChildContext();
const testData = {
    props: {
        allowedLocales: ['en', 'es', 'fr', 'it'],
        localeCode: 'en'
    },
    context: {
        store: {
            setState: jest.fn()
        },
        intl: intl
    }
};

describe('LocaleList', () => {
    beforeAll(() => {
        var mock = new MockAdapter(axios);
        mock.onPost('https://192.168.33.10/openmrs/ws/rest/v1/session').reply(200, {});
    });

    it('renders correctly', () => {
        const languageList = shallow(
            <LocaleList allowedLocales={testData.props.allowedLocales} localeCode={testData.props.localeCode} />,
            {context: testData.context}
        );

        expect(languageList.find('a').length).toBe(4);
        expect(shallowToJson(languageList)).toMatchSnapshot();
    });

    it('should change language', () => {
        const spyOnSetLanguage = jest.spyOn(LocaleList.prototype, 'setLanguage');
        const languageList = mount(
            <LocaleList allowedLocales={testData.props.allowedLocales} localeCode={testData.props.localeCode} />,
            {context: testData.context}
        );

        languageList
            .find('.locale')
            .at(1)
            .simulate('click');
        expect(spyOnSetLanguage).toHaveBeenCalledWith('es');
    });
});
