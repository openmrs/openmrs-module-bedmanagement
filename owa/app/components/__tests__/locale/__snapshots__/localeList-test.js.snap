// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleList renders correctly 1`] = `
<div>
  <small
    className="language-list"
  >
    <FormattedMessage
      id="CHOOSE_LANGUAGE"
      values={Object {}}
    />
    :  
    <span
      key="en"
    >
      <a
        className="active locale"
        href="javascript:void(0)"
        onClick={[Function]}
      >
        English
      </a>
    </span>
    <span
      key="es"
    >
       | 
      <a
        className="locale"
        href="javascript:void(0)"
        onClick={[Function]}
      >
        Español
      </a>
    </span>
    <span
      key="fr"
    >
       | 
      <a
        className="locale"
        href="javascript:void(0)"
        onClick={[Function]}
      >
        Français
      </a>
    </span>
    <span
      key="it"
    >
       | 
      <a
        className="locale"
        href="javascript:void(0)"
        onClick={[Function]}
      >
        Italiano
      </a>
    </span>
  </small>
</div>
`;
